.github/workflows/precommit.yaml

```diff
@@ -2,11 +2,8 @@ name: Precommit tests
 
 on:
   push:
-    branches:
-      - '*'
+    branches: [main, develop]
   pull_request:
```

Comment:
you'll want PR only on `main` and `develop` too (at least that's how it's setup everywhere else)